assets/blocks/button/edit-button.js

```diff
@@ -1,21 +1,53 @@
+import { useEffect } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { RichText } from '@wordpress/block-editor';
-import { useSelect } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
+
 import { getButtonProps, getButtonWrapperProps } from './button-props';
 import { ButtonBlockSettings } from './settings-button';
 
+/**
+ * Hook to set the default style if no style is defined.
+ *
+ * @param {string} clientId     Block client ID.
+ * @param {string} defaultStyle Default block style.
+ */
+const useDefaultStyle = ( clientId, defaultStyle ) => {
```

Comment:
Note that we also have `withDefaultStyle` — it adds `is-style-default` to the className, but it could be expanded to support a parameter:
https://github.com/Automattic/sensei/blob/master/assets/shared/blocks/settings.js#L80

Reply:
Hum! Good point!

I used it to have the class before updating the style, preventing the issue that shows a little time with the wrong style. https://github.com/Automattic/sensei/pull/3787/commits/c3b6305bc0a55fca165d678d16d39fe173addf03 https://github.com/Automattic/sensei/pull/3787/commits/5f211aa8822e999c70ac59a046dcc675683b9db1

But I also kept the hook (moving it to the settings file too), otherwise, we'd have to define the default style in 2 different places (for the editor and the frontend) https://github.com/Automattic/sensei/pull/3787/commits/d91b52ae964ad23417ad80ca1cae3593f517550a